### DISCUSSION TITLE
tarteaucitron.init doit s'executer avant l'événement load ?

### DISCUSSION THREAD

**Discussion by antoinegoutagny** [AmauriC/tarteaucitron.js#756] [Q&A]
Je constate que dans certains cas (notamment au rafraichissement de la page après une période d'inactivité), le script ne marche pas (pas de picto ni de popin). 

L'activation du script (méthode `loadEvents`) ne se fait que sur l'évènement `window.load`. L'écouteur est lui-même attaché dans la méthode `init`. Si `load` a lieu avant l'appel à `init`, le script ne fait rien. Est-ce un choix d'implémentation ? J'imagine que c'est pour ne pas ne pas alourdir le chargement.

Je précise que nos contraintes font que le script _tarteaucitron.js_ n'est pas appelé directement dans le html, mais dynamiquement dans notre propre script, ce qui explique probablement le retard à l'allumage. Je vais probablement rajouter une méthode publique pour forcer l'activation.

**Comment by AmauriC** [comment]
Si le script tarteaucitron est chargé en différé, il est probable que tarteaucitron.init() soit appelé avant que la librairie soit dispo et donc échoue.

**Reply by antoinegoutagny** [reply]
L'appel à `init` se fait bien évidemment après le load du script lui-même. Le problème n'est pas de l'appeler trop tôt, mais trop tard, après l'evenement `window.load` qui doit déclencher `loadEvent`.

Edit: pas besoin de rajouter une nouvelle méthode, j'appelle directement `tarteaucitron.initEvents.loadEvent`.

**Reply by AmauriC** [reply]
ok, merci pour le retour : )
